plugins/flashrom/fu-plugin-flashrom.c

```diff
@@ -143,32 +145,65 @@ fu_plugin_flashrom_device_set_hwids(FuPlugin *plugin, FuDevice *device)
 }
 
 static gboolean
-fu_plugin_flashrom_coldplug(FuPlugin *plugin, GError **error)
+fu_plugin_flashrom_add_device(FuPlugin *plugin, const gchar *region, GError **error)
 {
 	FuContext *ctx = fu_plugin_get_context(plugin);
-	const gchar *dmi_vendor;
-	g_autoptr(FuDevice) device = fu_flashrom_device_new(ctx);
+	const gchar *dmi_vendor = fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_BIOS_VENDOR);
 
-	fu_device_set_name(device, fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_PRODUCT_NAME));
-	fu_device_set_vendor(device, fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_MANUFACTURER));
+	g_autoptr(FuDevice) device = fu_flashrom_device_new(ctx, region);
 
-	/* use same VendorID logic as with UEFI */
-	dmi_vendor = fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_BIOS_VENDOR);
-	if (dmi_vendor != NULL) {
-		g_autofree gchar *vendor_id = g_strdup_printf("DMI:%s", dmi_vendor);
-		fu_device_add_vendor_id(FU_DEVICE(device), vendor_id);
-	}
-	fu_plugin_flashrom_device_set_version(plugin, device);
-	fu_plugin_flashrom_device_set_hwids(plugin, device);
-	fu_plugin_flashrom_device_set_bios_info(plugin, device);
-	if (!fu_device_setup(device, error))
+	if (g_strcmp0(region, "bios") == 0) {
+		fu_device_set_name(device,
+				fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_PRODUCT_NAME));
+		fu_device_set_vendor(device,
+				fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_MANUFACTURER));
+
+		/* use same VendorID logic as with UEFI */
+		if (dmi_vendor != NULL) {
+			g_autofree gchar *vendor_id = g_strdup_printf("DMI:%s", dmi_vendor);
+			fu_device_add_vendor_id(FU_DEVICE(device), vendor_id);
+		}
+		fu_plugin_flashrom_device_set_version(plugin, device);
+		fu_plugin_flashrom_device_set_hwids(plugin, device);
+		fu_plugin_flashrom_device_set_bios_info(plugin, device);
+		if (!fu_device_setup(device, error))
+			return FALSE;
+	} else if (g_strcmp0(region, "me") == 0) {
+		// XXX: should there be a hierarchy of devices?
+		// XXX: always create device but mark as FWUPD_DEVICE_FLAG_LOCKED if ME is locked?
```

Comment:
I think you need to use `FWUPD_DEVICE_FLAG_LOCKED` somwehere in this PR.  Probably on whatever device is to be locked/unlocked.